v2/tools/generator/internal/config/property_configuration.go

```diff
@@ -0,0 +1,109 @@
+//
+// ┌──────────────────────────┐       ┌────────────────────┐       ┌──────────────────────┐       ┌───────────────────┐       ╔═══════════════════════╗
+// │                          │       │                    │       │                      │       │                   │       ║                       ║
+// │ ObjectModelConfiguration │───────│ GroupConfiguration │───────│ VersionConfiguration │───────│ TypeConfiguration │───────║ PropertyConfiguration ║
+// │                          │1  1..n│                    │1  1..n│                      │1  1..n│                   │1  1..n║                       ║
+// └──────────────────────────┘       └────────────────────┘       └──────────────────────┘       └───────────────────┘       ╚═══════════════════════╝
+//
+type PropertyConfiguration struct {
+	name             string
+	renamedTo        *string
+	armReference     *bool
+	usedArmReference bool
+}
+
+// NewPropertyConfiguration returns a new (empty) property configuration
+func NewPropertyConfiguration(name string) *PropertyConfiguration {
+	return &PropertyConfiguration{
+		name: name,
+	}
+}
+
+// ARMReference looks up a property to determine whether it may be an ARM reference or not.
+func (pc *PropertyConfiguration) ARMReference() (bool, bool) {
+	if pc.armReference == nil {
+		return false, false
+	}
+
+	pc.usedArmReference = true
+	return *pc.armReference, true
+}
+
+// SetARMReference configures specifies whether this property is an ARM reference or not
+func (pc *PropertyConfiguration) SetARMReference(isARMRef bool) *PropertyConfiguration {
+	pc.armReference = &isARMRef
+	return pc
+}
+
+// FindUnusedARMReferences returns a slice listing any unused ARMReference configuration
+func (pc *PropertyConfiguration) FindUnusedARMReferences() []string {
+	var result []string
+	if pc.armReference != nil && !pc.usedArmReference {
+		msg := fmt.Sprintf("property %s:%t", pc.name, *pc.armReference)
+		result = append(result, msg)
+	}
+
+	return result
+}
+
+// SetRenamedTo configures what this property is renamed to in the next version of the containing type
+func (pc *PropertyConfiguration) SetRenamedTo(renamedTo string) *PropertyConfiguration {
+	pc.renamedTo = &renamedTo
+	return pc
+}
+
+// UnmarshalYAML populates our instance from the YAML.
+// The slice node.Content contains pairs of nodes, first one for an ID, then one for the value.
+func (pc *PropertyConfiguration) UnmarshalYAML(value *yaml.Node) error {
+	if value.Kind != yaml.MappingNode {
+		return errors.New("expected mapping")
+	}
+
+	var lastId string
+	for i, c := range value.Content {
+		// Grab identifiers and loop to handle the associated value
+		if i%2 == 0 {
+			lastId = strings.ToLower(c.Value)
+			continue
+		}
+
+		if lastId == "$renamedto" && c.Kind == yaml.ScalarNode {
+			pc.SetRenamedTo(c.Value)
+			continue
+		}
+
+		if lastId == "$armreference" && c.Kind == yaml.ScalarNode {
```

Comment:
Same question about case sensitivity here and below

Reply:
Same answer - I think configuration files should be a little forgiving.